File: test-suite/src/test/java/example/micronaut/OllamaChatModelBuilderListener.java
package example.micronaut;

import dev.langchain4j.model.ollama.OllamaChatModel;
import io.micronaut.context.event.BeanCreatedEvent;
import io.micronaut.context.event.BeanCreatedEventListener;
import org.jspecify.annotations.NonNull;
import jakarta.inject.Singleton;

@Singleton
class OllamaChatModelBuilderListener
    implements BeanCreatedEventListener<OllamaChatModel.OllamaChatModelBuilder> {
    @Override
    public OllamaChatModel.OllamaChatModelBuilder onCreated(
        @NonNull BeanCreatedEvent<OllamaChatModel.OllamaChatModelBuilder> event) {
        OllamaChatModel.OllamaChatModelBuilder builder = event.getBean();
        builder.temperature(0.0);
        return builder;
    }
}

File: test-suite-python/src/test/python/example/micronaut/OllamaChatModelBuilderListener.py
from dev.langchain4j.model.ollama import OllamaChatModel
from jakarta.inject import Singleton
from micronaut.context.event import BeanCreatedEvent, BeanCreatedEventListener


@Singleton
class OllamaChatModelBuilderListener(
    BeanCreatedEventListener[OllamaChatModel.OllamaChatModelBuilder]
):
    def onCreated(
        self, event: BeanCreatedEvent[OllamaChatModel.OllamaChatModelBuilder]
    ) -> OllamaChatModel.OllamaChatModelBuilder:
        builder = event.getBean()
        builder.temperature(0.0)
        return builder
